### PR TITLE
Update All deps to adapt to Assemblyscript v0.25

### PR DESCRIPTION
**Pull request by yjhmelody**
Now the host function is too new to be supported in old substrate contract node.
But now `start` wasm function is not supported now by substrate contract. See https://github.com/paritytech/polkadot-sdk/issues/116

So we need an another PR to rollback host functions aligned to polkadot-v0.9.32. Or just wait for upstream to re-support it.

@iamizumi

### CODE REVIEW

File: examples/asconfig.json [hunk 1]
```diff
@@ -1,13 +1,29 @@
...
+        "importMemory": true,
+        "initialMemory": 2,
+        "maximumMemory": 16,
+        "noExportMemory": true,
+        "runtime": "stub",
+        "use": "abort=",
+        "disable": ["Sign-extension"]
     }
-}
+  }
```
Comment (iamizumi) [inline examples/asconfig.json:29]:
```suggestion
}
```

File: ts-packages/transform/src/metadata/generator.ts [hunk 2]
```diff
@@ -36,27 +35,25 @@ import {
...
-} from ".";
-import debug from "debug";
-import { uniqBy } from "lodash";
-import { ArrayTypeInfo } from "./typeInfo";
+    ArrayTypeInfo,
+} from "./index.js";
 
 export const LANGUAGE = `Ask! ${ASK_VERSION}`;
-export const COMPILER = `asc ${version}`;
+export const COMPILER = `asc-0.24`;
```
Comment (iamizumi) [inline ts-packages/transform/src/metadata/generator.ts:56]:
```suggestion
export const COMPILER = `asc-0.25`;
```

File: ts-packages/transform/src/metadata/typeInfo.ts [hunk 3]
```diff
@@ -1,5 +1,9 @@
 import * as metadata from "ask-contract-metadata";
-import { Field, Type } from "visitor-as/as";
+import {
+    // TODO:
+    // Field,
```
Comment (iamizumi) [inline ts-packages/transform/src/metadata/typeInfo.ts:4]:
```suggestion
```

File: ts-packages/transform/src/metadata/typeResolver.ts [hunk 4]
```diff
@@ -1,28 +1,32 @@
...
+import debug from "debug";
+import { PrimitiveType } from "ask-contract-metadata";
 import {
     ClassPrototype,
     FunctionPrototype,
     Class,
     Type,
     ElementKind,
-    Field,
+    // Field,
```
Comment (iamizumi) [inline ts-packages/transform/src/metadata/typeResolver.ts:10]:
```suggestion
```

**Comment by yjhmelody** [comment]
Let's merge it soon, though it could not work recently until upstream fix contract problem.

**Review by lambda-world — APPROVED** [review]

**Comment by yjhmelody** [comment]
Let's move on.

**Pull request merged**